### ISSUE TITLE
Move InnerProductTypes out of base to proof_primitive

### ISSUE THREAD

**Issue by stuarttimwhite** [spaceandtimelabs/sxt-proof-of-sql#232]
# Background and Motivation
**NOTE**: https://github.com/spaceandtimelabs/sxt-proof-of-sql/issues/230 should be completed before this ticket can be done. Additionally, be aware that some of the code changes for this issue are likely to conflict with the code changes in the following issue:
https://github.com/spaceandtimelabs/sxt-proof-of-sql/issues/234

Currently, we have reference to 3 types that we want removed from proof-of-sql/src/base:
1. Curve25519Scalar
2. RistrettoPoint
3. InnerProductProof

 Any references to these types should be moved to the proof_primitive directory in its own directory named "inner_product". Ultimately, we are trying to remove all cyclic dependencies caused by these types.

# Changes Required

Each of these should be a separate PR.
- [ ] Move Curve25519Scalar and its unit tests to new directory proof_primitive/inner_product.
- [ ] Move RistrettoPoint and its unit tests to inner_product.
- [ ] Move InnerProductProof and its unit tests to inner_product.
- [ ] If there is any reference to any of these types remaining (most likely within a test), either move those to inner_product or replace with one of TestScalar, NaiveCommitment, or TestEvaluationProof.

**Comment by varshith257** [comment]
/attempt #232

Just waiting the issue #230 to be completed

**Comment by tmahmood** [comment]
Hey, @JayWhite2357 is this task open? I would like to give it a try

**Comment by JayWhite2357** [comment]
I believe it is open. Some of the parts may be partially completed.
@stuarttimwhite can confirm

**Comment by tmahmood** [comment]
Hey @JayWhite2357 

I have submitted the initial changes, please review.

**Comment by tmahmood** [comment]
@JayWhite2357 
Waiting for the review. Once this one is approved, I can get on with the other 3? Let me know what should I do.

**Comment by JayWhite2357** [comment]
@tmahmood I added a quick review.

**Comment by tmahmood** [comment]
@JayWhite2357 , @stuarttimwhite

Noting here,

TestScalar is only available in tests, So during doctests, it's not available.

So if I convert the doctests to use TestScalar, it will not find TestScalar and fail.

If I wrap it in a function, mark it as #[test] it also fails with this, [warning](https://rust-lang.github.io/rust-clippy/master/index.html#test_attr_in_doctest)

So, what should be done about this?
